pre_commit_hooks/update_commit_msg.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, print_function, unicode_literals
+import sys
+import optparse
+from invoke import run
+
+import re
+
+
+class UpdateCommitMsg(object):
+    """
+    This hook saves developers time by prepending ticket numbers to commit-msgs.
+    For this to work the following two conditions must be met:
+        - The ticket format regex specified must match.
+        - The branch name format must be <ticket number>_<rest of the branch name>
+    """
+
+    def __init__(self):
+        self.errors = 0
+        self.branch = self.get_branch_name()
+
+    def check_file(self, filename, regex):
+        self.current_filename = filename
+
+        with open(filename[0], 'r+') as fd:
+            contents = fd.readlines()
+            commit_msg = contents[0]
+            if not re.search(regex, commit_msg):
+                # Check if we can grab ticket from branch name.
+                # If not, do  nothing.
+                if re.search(regex, self.branch):
+                    ticket = self.branch.split('_')[0]
+                    new_commit_msg = '{} {}'.format(ticket, commit_msg)
+                    fd.seek(0)
+                    fd.write(new_commit_msg)
+                    fd.truncate()
+
+    def get_branch_name(self):
+        # Only git support for right now.
+        return run("git rev-parse --abbrev-ref HEAD").stdout
```

Comment:
```
subprocess.check_output(('git', 'rev-parse', '--abbref-ref', 'HEAD')).decode('UTF-8')
```

there's also potentially a failure mode when you're not on a branch that might be worth handling or ignoring (?)